src/Swift.Bindings/src/TypeDatabase/TypeRecord.cs

```diff
@@ -5,6 +5,29 @@
 
 namespace BindingsGeneration;
 
+/// <summary>
+/// Represents type's flags.
+/// </summary>
+[Flags]
+public enum TypeRecordFlags
+{
+    None = 0,
+    Frozen = 1 << 0,
+    HeapAllocated = 1 << 1,
+}
+
+/// <summary>
+/// Represents a type kind.
+/// </summary>
+public enum TypeRecordKind
+{
+    Struct,
+    Enum,
+    Class,
+    Protocol,
+    Tuple
```

Comment:
Do we need to place tuples in the type database? Arent there always constructed from other types?

Reply:
Void is the only case and it could be special-handled.